Add MongoDb Trigger node that works using change streams

DaKyzen opened the pull request:
A trigger node for MongoDb that outputs the change event from a change stream as a JSON. There is an option to output either the full document or only fields which have been changed.

DaKyzen committed:
- Create MongoDbTrigger node

DaKyzen committed:
- Refactor credentials to use same method as mongo node
- Refactor description into its own file

DaKyzen committed:
- Remove pipeline options. JSON can be modified in other nodes
- Add boolean for getting the full document

DaKyzen committed:
- Add the trigger node to package.json

DaKyzen committed:
- Remove unused imports

DaKyzen committed:
- Add check for collection name if it exists. Previously if a collection name that did not exist was entered, the node would be stuck

DaKyzen committed:
- Flatten json being returned
- Create type and interface to be used when flattening json
- Return an empty array when the collection name is not found

DaKyzen committed:
- Add useNewUrlParser and useUnifiedTopology options to constructor of client

DaKyzen committed:
- Commit package-lock.json

michael-radency commented:
Hello @DaKyzen , can you please explain how to correctly setup this node or maybe share some exemplary workflow?
Also this isn't n8n trigger node, the name is misleading, you should rename it or, if it possible, change it to trigger node, here some information that can be useful https://docs.n8n.io/nodes/creating-nodes/create-trigger-node.html#prerequisites

DaKyzen committed:
Merge branch 'n8n-io:master' into master

DaKyzen committed:
- Fix trigger node not appearing under the trigger section
- Update the default name to be the same as the display name

DaKyzen commented:
Hello @michael-radency,

The node is for detecting changes to a MongoDb collection and outputting the change details.

Setting up the node is the same as the current MongoDb node in terms of entering the credentials. As for the parameters, there's only the collection field and the Full Json. The collection field is for specifying what collection the node should be watching, while the full documnet switch specifies whether or not to show only the fields that have changed or all the fields of the record. Hopefully the two screenshots make things clearer.

![Sample workflow](https://user-images.githubusercontent.com/73026342/158066224-b7fed701-19f3-4e82-9319-90b79afee38c.PNG)

![Table](https://user-images.githubusercontent.com/73026342/158066216-caaad5a4-315b-4d4c-9a44-08cfc8b0200d.PNG)

As for the node not being a trigger node, can you elaborate further? I believe that I followed the instructions on creating a trigger node from your link. There was a bug that I just noticed where my trigger node did not appear under the trigger section in the node list which I fixed. I've also updated the name to be more specific to change streams.

michael-radency committed:
:hammer: fixed trigger node to work as trigger, small improvements

michael-radency commented:
Hello  @DaKyzen, thank you for your contribution, I did made some fixes to make it work as a trigger node, also I'm wondering why do you flattening JSON object? for now I did remove that, if this need to be done please consider adding additional option for that
